core/src/main/java/jenkins/security/SuppressionFilter.java

```diff
@@ -0,0 +1,160 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.security;
+
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+import hudson.util.PluginServletFilter;
+import jenkins.model.Jenkins;
+import org.acegisecurity.AccessDeniedException;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.HttpResponses;
+import org.kohsuke.stapler.Stapler;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
+
+@Restricted(NoExternalUse.class)
+public class SuppressionFilter implements Filter {
+
+    private static final Logger LOGGER = Logger.getLogger(SuppressionFilter.class.getName());
+
+    @Initializer(after = InitMilestone.STARTED)
+    public static void init() throws ServletException {
+        HttpResponses.setErrorDetailsFilter((code, cause) -> showStackTrace());
+        PluginServletFilter.addFilter(new SuppressionFilter());
+    }
+
+    public void init(FilterConfig filterConfig) {
+        // no-op
+    }
+
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        try {
+            chain.doFilter(request, response);
+        } catch (Exception e) {
+            if (containsAccessDeniedException(e)) {
+                throw e;
+            }
+            if (Stapler.isSocketException(e)) {
+                return;
+            }
+
+            if (showStackTrace()) {
+                if (e instanceof IOException || e instanceof ServletException || e instanceof RuntimeException) {
+                    // thing we can throw without wrapping
+                    throw e;
+                }
+                throwServletException(e);
+            } else if (request instanceof HttpServletRequest && response instanceof HttpServletResponse) {
+                String errorId = logException((HttpServletRequest) request, e);
+
+                respondWithSimpleErrorPage((HttpServletResponse) response, errorId);
+            }
+        }
+    }
+
+    private void respondWithSimpleErrorPage(HttpServletResponse response, String errorId) throws ServletException {
+        try {
+            response.setStatus(SC_INTERNAL_SERVER_ERROR);
```

Comment:
check `isCommitted()` first.  if the response is committed this will all just blow up and spam the log...
I expect many times that the response would be committed and thus this would be always doomed to fail.